WordPress/src/main/java/org/wordpress/android/models/usecases/BatchModerateCommentsUseCase.kt

```diff
@@ -0,0 +1,124 @@
+import org.wordpress.android.fluxc.model.CommentStatus
+import org.wordpress.android.fluxc.model.CommentStatus.DELETED
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.CommentStore.CommentError
+import org.wordpress.android.fluxc.store.CommentStore.CommentErrorType.GENERIC_ERROR
+import org.wordpress.android.fluxc.store.CommentsStore.CommentsData.DoNotCare
+import org.wordpress.android.models.usecases.BatchModerateCommentsUseCase.ModerateCommentsAction
+import org.wordpress.android.models.usecases.BatchModerateCommentsUseCase.ModerateCommentsAction.OnModerateComments
+import org.wordpress.android.models.usecases.BatchModerateCommentsUseCase.ModerateCommentsState.Idle
+import org.wordpress.android.models.usecases.BatchModerateCommentsUseCase.Parameters.ModerateCommentsParameters
+import org.wordpress.android.models.usecases.CommentsUseCaseType.BATCH_MODERATE_USE_CASE
+import org.wordpress.android.usecase.FlowFSMUseCase
+import org.wordpress.android.usecase.UseCaseResult
+import org.wordpress.android.usecase.UseCaseResult.Failure
+import javax.inject.Inject
+
+class BatchModerateCommentsUseCase @Inject constructor(
+    moderateCommentsResourceProvider: ModerateCommentsResourceProvider
+) : FlowFSMUseCase<ModerateCommentsResourceProvider, ModerateCommentsParameters, ModerateCommentsAction, DoNotCare,
+        CommentsUseCaseType, CommentError>(
+        resourceProvider = moderateCommentsResourceProvider,
+        initialState = Idle
+) {
+    override suspend fun runInitLogic(parameters: ModerateCommentsParameters) {
+        manageAction(OnModerateComments(parameters))
+    }
+
+    sealed class ModerateCommentsState
+        : StateInterface<ModerateCommentsResourceProvider, ModerateCommentsAction, DoNotCare, CommentsUseCaseType,
+            CommentError> {
+        object Idle : ModerateCommentsState() {
+            override suspend fun runAction(
+                resourceProvider: ModerateCommentsResourceProvider,
+                action: ModerateCommentsAction,
+                flowChannel: MutableSharedFlow<UseCaseResult<CommentsUseCaseType, CommentError, DoNotCare>>
+            ): StateInterface<ModerateCommentsResourceProvider, ModerateCommentsAction, DoNotCare, CommentsUseCaseType,
+                    CommentError> {
+                val commentsStore = resourceProvider.commentsStore
+                return when (action) {
+                    is OnModerateComments -> {
+                        val parameters = action.parameters
+                        withContext(resourceProvider.bgDispatcher) {
+                            val deferredList = parameters.remoteCommentIds.map {
+                                async {
+                                    val commentBeforeModeration = commentsStore.getCommentByLocalSiteAndRemoteId(
+                                            parameters.site.id,
+                                            it
+                                    ).first()
```

Comment:
If I'm not wrong, in principle here the list can be null. I would maybe do a `firstOrNull` and emit an error in case of null. wdyt?

Reply:
Good idea 👍